File: app/src/main/java/com/arielfaridja/ezrahi/UI/Main/MainActivityViewModel.java
package com.arielfaridja.ezrahi.UI.Main;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.arielfaridja.ezrahi.data.DataRepo;
import com.arielfaridja.ezrahi.data.DataRepoFactory;
import com.arielfaridja.ezrahi.entities.Activity;
import com.arielfaridja.ezrahi.entities.Latlng;
import com.arielfaridja.ezrahi.entities.User;

public class MainActivityViewModel extends ViewModel {


    private User user;
    Context context;
    DataRepo dataRepo = DataRepoFactory.getInstance();
    Intent intent;

    MutableLiveData<Boolean> isSignIn = new MutableLiveData<>(null);
    private Activity activity;

    public MainActivityViewModel() {
        super();
        getIsSignIn().setValue(dataRepo.user_isSignedIn());

    }

    public Activity getActivity() {
        return activity;
    }

    public void setActivity(Activity activity) {
        this.activity = activity;
    }

    public MutableLiveData<Boolean> getIsSignIn() {
        return isSignIn;
    }

    public void setIsSignIn(MutableLiveData<Boolean> isSignIn) {
        this.isSignIn = isSignIn;
    }

    void init(Intent intent) {
        initUser(intent);
    }

    void initUser(Intent intent) {
        this.user = new User(intent.getStringExtra("id"),
                intent.getStringExtra("firstName"),
                intent.getStringExtra("lastName"),
                intent.getStringExtra("email"),
                intent.getStringExtra("phone"),
                new Latlng(intent.getDoubleExtra("longitude", 32.0D), intent.getDoubleExtra("latitude", 34.0D)));
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public void isUserSignedIn() {
        if (!dataRepo.user_isSignedIn())
            isSignIn.setValue(false);
        else
            isSignIn.setValue(true);
    }

    public void loadUserDatafromSP(SharedPreferences sp) {
        user = new User();
        user.setId(sp.getString("uId", ""));
        user.setFirstName(sp.getString("FirstName", ""));
        user.setLastName(sp.getString("LastName", ""));
        user.setLocation(new Latlng(
                Double.parseDouble(sp.getString("Latitude", "0.0")),
                Double.parseDouble(sp.getString("Longitude", "0.0"))));
        user.setPhone(sp.getString("Phone", ""));
        user.setEmail(sp.getString("Email", ""));
    }

    public void loadActivityDatafromSP(SharedPreferences sp) {
        activity = new Activity();
    }
}

File: app/src/main/java/com/arielfaridja/ezrahi/UI/Main/MainActivityViewModel.kt
package com.arielfaridja.ezrahi.UI.Main

import android.content.Intent
import android.content.SharedPreferences
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.arielfaridja.ezrahi.data.DataRepoFactory
import com.arielfaridja.ezrahi.data.IDataRepo
import com.arielfaridja.ezrahi.entities.Activity
import com.arielfaridja.ezrahi.entities.Latlng
import com.arielfaridja.ezrahi.entities.User

class MainActivityViewModel : ViewModel() {
    private val dataRepo: IDataRepo = DataRepoFactory.getInstance()
    private var intent: Intent? = null
    private var isSignIn: MutableLiveData<Boolean> = MutableLiveData(null)
    private var currentActivity: MutableLiveData<Activity> = MutableLiveData(dataRepo.activity_getCurrent())
    private var user: User? = null
    private var userLiveData: MutableLiveData<User?> = MutableLiveData(null)

    init {
        isSignIn.value = dataRepo.user_isSignedIn()
        user = dataRepo.user_getCurrent()
        userLiveData.value = user
    }

    fun getIsSignIn(): MutableLiveData<Boolean> = isSignIn
    fun setIsSignIn(isSignIn: MutableLiveData<Boolean>) { this.isSignIn = isSignIn }
    fun getActivity(): Activity? = currentActivity.value
    fun setActivity(activity: Activity) { currentActivity.value = activity }
    fun getCurrentActivity(): MutableLiveData<Activity> = currentActivity
    fun getCurrentUsersActivities() = dataRepo.activity_getAllByCurrentUser()
    fun getUser(): User? = user
    fun setUser(user: User) {
        this.user = user
        this.userLiveData.value = user
    }
    fun getUserLiveData(): MutableLiveData<User?> = userLiveData

    fun init(intent: Intent) {
        initUser(intent)
    }

    fun initUser(intent: Intent) {
        user = User(
            intent.getStringExtra("id") ?: "",
            intent.getStringExtra("first_name") ?: "",
            intent.getStringExtra("last_name") ?: "",
            intent.getStringExtra("phone") ?: "",
            intent.getStringExtra("email") ?: "",
            Latlng(
                intent.getDoubleExtra("latitude", 34.0),
                intent.getDoubleExtra("longitude", 32.0)
            )
        )
        userLiveData.value = user
    }

    fun isUserSignedIn() {
        isSignIn.value = dataRepo.user_isSignedIn()
    }


    fun loadActivityDataFromSP(sp: SharedPreferences) {
        val actId = sp.getString("actId", "") ?: ""
        if (actId.isNotEmpty()) {
            dataRepo.activity_setCurrent(actId)
            Thread {
                while (dataRepo.activity_getCurrent() == null) {}
                currentActivity.postValue(dataRepo.activity_getCurrent())
            }.start()
            // TODO: populate activity
        }
    }

    fun loadUserDataFromSP(sp: SharedPreferences) {
        user = User()
        user?.id = sp.getString("uId", "") ?: ""
        user?.firstName = sp.getString("FirstName", "") ?: ""
        user?.lastName = sp.getString("LastName", "") ?: ""
        user?.location = Latlng(
            (sp.getString("Latitude", "0.0") ?: "0.0").toDouble(),
            (sp.getString("Longitude", "0.0") ?: "0.0").toDouble()
        )
        user?.phone = sp.getString("Phone", "") ?: ""
        user?.email = sp.getString("Email", "") ?: ""
        userLiveData.value = user
    }
}
